features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/recorder/RecordingTimeBank.kt

```diff
@@ -0,0 +1,49 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.sessionreplay.internal.recorder
+
+import java.util.concurrent.TimeUnit
+import kotlin.math.min
+
+/**
+ * Implementation of [TimeBank], it maintains a time balance which can be consumed for session replay recording.
+ * Everytime the session replay recording task tries to execute, [updateAndCheck] to increase the time balance
+ * according to the time passed since last call, and check if the balance is positive to execute the task.
+ */
+internal class RecordingTimeBank(
+    private val maxTimeBalancePerSecondInMs: Long = DEFAULT_MAX_TIME_BALANCE_PER_SEC_IN_MS
+) : TimeBank {
+
+    @Volatile
+    private var recordingTimeBalanceInNano = TimeUnit.MILLISECONDS.toNanos(maxTimeBalancePerSecondInMs)
+    private var lastStartTime: Long = 0
+
+    // The time to increase 1 nano second balance in the bank. For example the max balance per second is 100ms,
```

Comment:
```suggestion
    // The time to increase 1 nanosecond balance in the bank. For example the max balance per second is 100ms,
```

Also I'm still not clear with the description overall. Why do we need this property?

Reply:
This is the factor of balance, bigger it is, faster the balance will increase by time.
I removed it as the class property, and changed it to a local variable, and also update the comment to make it more clear, please check if it is more understandable to you?

Reply:
Well for the reason why we need this variable, is that I try to avoid to have a super long mathematic formula in the code, so I add this intermediate variable with name to make it more understandable.


This `balanceFactor` is calculated from the parameter in the constructor `maxTimeBalancePerSecondInMs`, for example if `maxTimeBalancePerSecondInMs`=100ms, it means we should only allow 100ms per second to execute the task that we want. So in this time bank, it also means that we can have maximum 100ms in the balance, and to reach the max (100ms) from 0 we need 1000ms, which means the increasing factor is 0.1.


With this factor, we can easily calculate the balance increasing by the given time interval, for example last time when the balance update is at `t0`, current time is `t1`, then the balance we should increase should be `(t1 - t0)*factor`.

Don't hesitate to tell me if it is still confusing,  the goal is to make the code and the document understandable for every one.

Reply:
My question is not the formula, my question is why we need this. Why do we need the factor and why we simply not matching bank with execution time by using 1:1 ratio.

Reply:
Document updated